### DISCUSSION TITLE
I can't get my authentication code on github mobile

### DISCUSSION THREAD

**Discussion by CoolwindHF** [community/community#81526] [Other Feature Feedback, Questions, & Ideas] [answered]
when i login in github, the page says that i should verify my 2FA and open my github mobile to get a authentication code, but i can't get it.

**Comment by SMGolamAlif** [comment, marked as answer]
Getting stuck with 2FA can be a headache. If you're having trouble receiving the authentication code on your GitHub mobile app, here are a few quick checks:

Network Connection: Ensure your mobile device has a stable internet connection. Sometimes a wonky connection can delay code delivery.

App Permissions: Double-check if GitHub has the necessary permissions on your mobile device. Sometimes adjusting these settings can do the trick.

Device Clock: Ensure the clock on your mobile device is accurate. A time discrepancy can mess with the authentication process.

Check Spam Folder: Sometimes, these codes sneak into the spam folder. Give it a quick look, just in case.

If none of these work, maybe try restarting your mobile app or logging in through a different method (text, call, etc.). If the problem persists, reaching out to GitHub support might be your best bet. They're usually pretty quick at helping out with these issues.

**Reply by 168ranew-cmd** [reply]
81526

**Comment by DNin01** [comment]
As part of the process of setting up 2FA, you will have at least configured an authenticator app such as Google Authenticator or to receive codes via text messages, plus recovery codes. If you're having trouble and those options are available, select one of them under "Having trouble?" on the authentication prompt.
![GitHub Sudo mode prompt highlighting alternative option links](https://github.com/community/community/assets/106490990/ea970471-8bf7-4c18-8b34-4b37c762fc13)

Also check that you have selected the correct method. An authenticator app just means an app that uses time-based one-time passwords (TOTPs), not GitHub Mobile, which is different. To use GitHub Mobile instead of a TOTP app to authenticate, click "Use GitHub Mobile" on the prompt. If you like to use GitHub Mobile by default for authentication, you can set it as your preferred method in Settings → Password and authentication → Preferred 2FA method.

**Reply by rudronix** [reply]
I have lost access to my GitHub account due to accidentally deleting the authentication app (e.g., Google Authenticator/Authy) from my phone. As a result, I am unable to enter the 2FA code required to log in.

I still have access to my email address associated with the account, and I can verify my identity in any way needed. Could you please help me recover access to my account or reset the two-factor authentication?

GitHub Username: [itzaryan008]
Registered Email: [itzaryan404@yahoo.com]

Thank you for your assistance.

Best regards,
[itzaryan008]

**Reply by frowingo** [reply]
same problem dude. I try to reach customer success and I wanna solve the problem but account is gg. this my second account.

**Comment by ghostinhershell** [comment]
Hey there!  👋

Thanks for posting in the GitHub Community, @CoolwindHF! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

With that being said, I do see that both @SMGolamAlif and @DNin01 have both provided possible solutions for your question. If either one doesn't resolve the issue you're seeing on your end, let me know and I can take a look here. 

I've gone ahead and moved this to the correct category for you. Good luck!

**Reply by sudo-update** [reply]
I have the same issue unfortunately. These solutions didn't work for me.  The website tells me I need to enter the 2FA code, and my app on my phone does not have an option anywhere in it to get a 2FA code.  What am I supposed to do? There was an option to defer 2FA until tomorrow, but I'm going to run into the exact same issue again tomorrow...

**Reply by CoolwindHF** [reply]
my solution is use the authenticator plugin in Chrome, so that i can get the code whenever i want to login Github, although it isn't as convenient as using the mobile phone.

**Reply by candrew34** [reply]
> my solution is use the authenticator plugin in Chrome, so that i can get the code whenever i want to login Github, although it isn't as convenient as using the mobile phone.

Can you share the link to the plug-in?

**Reply by CoolwindHF** [reply]
You can try to search in Google plugin store with the key word "Authenticator", and you can see lots of similar plugins for chrome. BUT I need to login in Github at different devices, so now I use the phone app like Authenticator by Microsoft to complete my 2FA verification.

**Comment by rudronix** [comment]
I still have access to my email address associated with the account, and I can verify my identity in any way needed. Could you please help me recover access to my account or reset the two-factor authentication?

GitHub Username: [itzaryan008]
Registered Email: [itzaryan404@yahoo.com]

Thank you for your assistance.

Best regards,
[itzaryan008]

**Comment by losmurfs** [comment]
On iOS, this is the only way to get it to work consistently:

1. Open GitHub Mobile (If GitHub Mobile is already "running" in the background, interact with it in the foreground to make sure it isn't so asleep that it won't work when called on).
2. Put some other app in the foreground so that GitHub is running in the background
3. Start logging in via GitHub Mobile, don't start the authentication process until you have done steps 1 and 2.
4. Clicking on the notification you receive to authenticate via GitHub only works when GitHub is in the background, if GitHub is in the foreground, nothing will happen. From this point on, log in will be intuitive.

**Comment by Wholinator** [comment]
Same issue, no resolution. I thought enabling notifications might do it but still nothing at all! This is insane! It's bothering me every single time I attempt to access any github url anywhere and telling me I have to do this "to avoid potential lockout". Well then let me do it! The 2 digit codes work fine but the 6 digit ones just go into the void apparently. I've updated the app, enabled notifications, restarted my phone, refreshed and re-accessed the code-request page in different browsers, with extensions disabled and not a single thing can make the authenticator actually work! Why are you requiring us to do this when it doesn't work!

**Comment by danabadarneh** [comment]
Thanks for raising this issue. Authentication problems on GitHub Mobile can be really frustrating.
In my experience, checking notification permissions, internet connection, and time/date settings on the phone can help. Reinstalling the app or using backup authentication methods is also a good option.

Hope this gets resolved soon. Thanks to the team for looking into it!

**Comment by itsrakesshchauhan** [comment]
If the problem remain unsolved, just log out your profile from all devices and then reinstall the app and then try.

**Reply by ThosRTanner** [reply]
i find it hard to believe this problem is considered 'answered'. I'm lucky I have the authenticator app because even though github claims it's sent a message to my mobile app, my mobile app sits there like a lemon. moreover I can't see anywhere on my mobile app that would allow me to check just in case.
